example/timer/odp_timer_accuracy.c

```diff
@@ -876,36 +946,43 @@ static void cancel_periodic_timers(test_global_t *test_global)
 	}
 }
 
-static void run_test(test_global_t *test_global)
+static int run_test(void *arg)
 {
-	uint64_t burst, num, num_tmo, next_tmo;
-	uint64_t i, tot_timers;
+	test_global_t *test_global = (test_global_t *)arg;
 	odp_event_t ev;
 	odp_time_t time;
 	uint64_t time_ns, diff_ns;
 	odp_timeout_t tmo;
 	uint64_t tmo_ns;
 	timer_ctx_t *ctx;
-	test_stat_t *stat = &test_global->stat;
-	test_log_t *log = test_global->log;
+	test_shm_t *shm = test_global->shm;
+	test_log_t *log = shm->log;
 	enum mode_e mode = test_global->opt.mode;
-	double period_dbl = test_global->period_dbl;
-	odp_timer_pool_t tp = test_global->timer_pool;
+	uint64_t tot_timers = test_global->tot_timers;
+	int tid = odp_thread_id();
+	test_stat_t *stat = &shm->stat[tid];
 
 	memset(stat, 0, sizeof(*stat));
 	stat->nsec_before_min = UINT64_MAX;
 	stat->nsec_after_min = UINT64_MAX;
 
-	num      = 0;
-	next_tmo = 1;
-	num_tmo  = test_global->opt.num_warmup + test_global->opt.num;
-	burst    = test_global->opt.burst;
-	tot_timers = test_global->tot_timers;
+	odp_barrier_wait(&shm->barrier);
 
-	for (i = 0; i < tot_timers; i++) {
-		ev = odp_schedule(NULL, ODP_SCHED_WAIT);
+	double period_dbl = test_global->period_dbl;
+	odp_timer_pool_t tp = test_global->timer_pool;
+
+	while (1) {
+		if (odp_atomic_load_u64(&shm->events) >= tot_timers)
+			break;
```

Comment:
This check on every round and before schedule call adds latency (if tmo event is waiting in a queue). So, better to move it under "ev == INVALID" branch, and maybe schedule call could have e.g. a 100msec timeout instead of NO_WAIT.

The "Free current scheduler context" loop can be removed from the end of the function, if we break schedule loop only on EVENT_INVALID.